pysmarthashtag/api/utils.py

```
@@ -69,6 +116,82 @@ def generate_default_header(
+
+    timestamp = create_correct_timestamp()
+    nonce = str(uuid.uuid4()).upper()
+
+    # INTL requires URL-encoded params in signature for GET requests with special chars
+    url_encode_params = method.upper() == "GET"
+    sign = _create_sign(nonce, params, timestamp, method, url, body, use_intl=True, url_encode_params=url_encode_params)
+
+    # Get vehicle series from params if available (passed as _vehicle_series)
+    vehicle_series = params.pop("_vehicle_series", None) if params else None
+    if not vehicle_series:
+        raise ValueError("vehicle_series is required for INTL API requests. Pass it in params as '_vehicle_series'.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)